### PR TITLE
 cache max size metrics

### PR DESCRIPTION
**Pull request by khushbr**
*Issue #:* https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/issues/296

*Description of changes:*  Adding code change for collecting cache max size metrics for Field Data and Shard Request Cache.
```
sqlite> select * from Cache_MaxSize;
Field Data Cache|26214400.0|26214400.0|26214400.0|26214400.0
Shard Request Cache|80181985.0|80181985.0|80181985.0|80181985.0
sqlite>
```

Older PR : https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/pull/270

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/metrics/Cache_FieldData_Max_Size.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.metrics;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.AllMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.Metric;
+
+public class Cache_FieldData_Max_Size extends Metric {
+    public Cache_FieldData_Max_Size(long evaluationIntervalSeconds) {
+        super(AllMetrics.ShardStatsValue.CACHE_FIELDDATA_MAX_SIZE.name(), evaluationIntervalSeconds);
```
Comment (rguo-aws) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/metrics/Cache_FieldData_Max_Size.java:23]: change to .toString() ?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/metrics/Cache_Request_Max_Size.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.metrics;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.AllMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.Metric;
+
+public class Cache_Request_Max_Size extends Metric {
+    public Cache_Request_Max_Size(long evaluationIntervalSeconds) {
+        super(AllMetrics.ShardStatsValue.CACHE_REQUEST_MAX_SIZE.name(), evaluationIntervalSeconds);
```
Comment (rguo-aws) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/metrics/Cache_Request_Max_Size.java:23]: same as above

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metrics/AllMetrics.java [hunk 3]
```diff
@@ -129,6 +130,44 @@ public String toString() {
     }
   }
 
+  public enum CacheConfigDimension implements MetricDimension {
```
Comment (rguo-aws) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metrics/AllMetrics.java:133]: add java doc for this dimension enum?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/metrics/Cache_FieldData_Max_Size.java (hunk 1 above)
Comment (khushbr) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/metrics/Cache_FieldData_Max_Size.java:23]: Done, updated at all cache relevant files.

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/metrics/Cache_Request_Max_Size.java (hunk 2 above)
Comment (khushbr) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/metrics/Cache_Request_Max_Size.java:23]: Done, updated at all cache relevant files.

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metrics/AllMetrics.java (hunk 3 above)
Comment (khushbr) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/metrics/AllMetrics.java:133]: Added.

**Pull request merged**